metaflow/plugins/aws/awslambda/lambda_decorator.py

```
@@ -0,0 +1,208 @@
+
+
+class LambdaDecorator(StepDecorator):
+    """
+    The decorator is called @awslambda to avoid confusion, since "lambda" is a reserved
+    word in Python.
+    """
+
+    name = "awslambda"
+    defaults = {
+        "image": None,
+        "force_fetch_image": False,
+    }
+
+    package_url = None  # type: Optional[str]
+    package_sha = None  # type: Optional[str]
+    resource_attributes = {}  # type: Dict[str, str]
+    run_time_limit = None
+    ds_root = None  # type: Optional[str]
+
+    def step_init(
+        self,
+        flow,  # type: "FlowSpec"
+        graph,  # type : "FlowGraph"
+        step,  # type: str
+        decos,  # type: List[Decorator]
+        environment,  # type: "MetaflowEnvironment"
+        datastore,  # type: "MetaflowDataStore"
+        logger,  # type: "EventLogger"
+    ):
+        if datastore.TYPE != "s3":
+            raise LambdaException("The *@awslambda* decorator requires --datastore=s3.")
+
+        self.logger = logger
+        self.environment = environment
+        self.step = step
+        for deco in decos:
+            if isinstance(deco, CondaStepDecorator):
+                raise LambdaException(
+                    "*@awslambda* doesn't support using @conda. "
+                    "We recommend using custom docker images to manage"
+                    " dependencies instead"
+                )
+
+            if isinstance(deco, BatchDecorator):
+                raise LambdaException(
+                    "You can't use @batch and *@awslambda* at the same time"
+                )
+
+            if isinstance(deco, ResourcesDecorator):
+                for k, v in deco.attributes.items():
+                    if v != deco.defaults[k]:
+                        if k != "memory":
+                            raise LambdaException(
+                                "*@awslambda* only supports provisioning memory via @resources, not {r}".format(
+                                    r=k
```

I think rather than throwing an error, we can just ignore `cpu` etc. Makes it easier for folks to float between batch and lambda that way. Thoughts?

---

Hmm good point, but it may be confusing for some if CPU and (especially) GPU settings are silently ignored but memory is not.